bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-messages/chat-list/chat-list-item/component.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect } from 'react';
+import { layoutSelect, layoutSelectInput, layoutDispatch } from '/imports/ui/components/layout/context';
+import { ACTIONS, PANELS } from '/imports/ui/components/layout/enums';
+import { defineMessages, useIntl } from 'react-intl';
+import { Meteor } from 'meteor/meteor'
+import Styled from './styles';
+import UserAvatar from '/imports/ui/components/user-avatar/component';
+import Icon from '/imports/ui/components/common/icon/component';
+import { Input, Layout } from '/imports/ui/components/layout/layoutTypes';
+import { Chat } from './chatTypes';
+import { useShortcutHelp } from '/imports/ui/components/shortcut-help/useShortcutHelp'
+
+const intlMessages = defineMessages({
+  titlePublic: {
+    id: 'app.chat.titlePublic',
+    description: 'title for public chat',
+  },
+  unreadPlural: {
+    id: 'app.userList.chatListItem.unreadPlural',
+    description: 'singular aria label for new message',
+  },
+  unreadSingular: {
+    id: 'app.userList.chatListItem.unreadSingular',
+    description: 'plural aria label for new messages',
+  },
+});
+
+interface ChatListItemProps {
+  chat: Chat,
+}
+
+const CHAT_CONFIG = Meteor.settings.public.chat;
+const ROLE_MODERATOR = Meteor.settings.public.user.role_moderator;
+
+const PUBLIC_GROUP_CHAT_ID = CHAT_CONFIG.public_group_id;
+
+const ChatListItem = (props: ChatListItemProps) => {
+  const sidebarContent = layoutSelectInput((i: Input) => i.sidebarContent);
+  const idChatOpen = layoutSelect((i: Layout) => i.idChatOpen);
+  const layoutContextDispatch = layoutDispatch();
+
+  const { sidebarContentPanel } = sidebarContent;
+  const sidebarContentIsOpen = sidebarContent.isOpen;
+
+  const TOGGLE_CHAT_PUB_AK: Object | undefined = useShortcutHelp();
+  const {
+    chat,
+  } = props;
+
+  const isPublicGroupChat = (chat: Chat) => chat.chatId === PUBLIC_GROUP_CHAT_ID;
```

Comment:
declare the function out of the component, in the current way every render the functil will be re-declared